share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -134,45 +141,30 @@ default:
   tags: ["spack", "public", "medium", "x86_64"]
 
 .generate-aarch64:
-  extends: [ ".generate" ]
+  extends: [ ".base-job", ".generate" ]
   tags: ["spack", "public", "medium", "aarch64"]
 
-.pr-generate:
-  extends: [ ".pr", ".generate" ]
-
-.pr-generate-deprecated:
-  extends: [ ".pr", ".generate-deprecated" ]
-
-.pr-generate-aarch64:
-  extends: [ ".pr", ".generate-aarch64" ]
-
-.protected-generate:
-  extends: [ ".protected", ".generate" ]
-
-.protected-generate-deprecated:
-  extends: [ ".protected", ".generate-deprecated" ]
-
-.protected-generate-aarch64:
-  extends: [ ".protected", ".generate-aarch64" ]
-
 .build:
+  extends: [ ".base-job" ]
   stage: build
 
-.pr-build:
-  extends: [ ".pr", ".build", ".aws-pr-creds" ]
-
-.protected-build:
-  extends: [ ".protected", ".build", ".aws-protected-creds" ]
-
 protected-publish:
+  # Copy binaries from stack-specific mirrors to a root mirror
   stage: publish
-  extends: [ ".protected", ".aws-protected-creds" ]
+  only:
+  - /^develop$/
+  - /^releases\/v.*/
+  - /^v.*/
```

Comment:
should this be `only: [..., tags, ...]`?

Reply:
I'm not clear why.  Is that about the yaml list syntax?  Or is the point about not putting tags at the beginning or end of the list?

Reply:
I mean, can we avoid running this on branches starting with `v`, and instead only on tags?

Reply:
just a note, this is identical to the previous behavior. Note, this is running on the gitlab side, so PRs are all prefixed with `pr<#>_` and only develop and release branches are pushed without the special prefixes.

Reply:
okay, I don't have strong opinions, but it would be a bit cleaner to fix it at some point